src/pulumi-metadata.ts

```diff
@@ -260,24 +259,24 @@ export function getNativeType(
  * Recursively normalizes object types, with special handling for JSON types (which should not be normalized)
  *
  * @param key the property key as a list (including parent property names for nested properties)
- * @param value the value to normalize
+ * @param value the value to normalize; this should not contain eventual types like Promise and pulumi.Output
  * @param cfnType The CloudFormation resource type being normalized (e.g. AWS::S3::Bucket). If no value
  * is provided then property conversion will be done without schema knowledge
  * @param pulumiProvider The pulumi provider to read the schema from. If `cfnType` is provided then this defaults
  * to PulumiProvider.AWS_NATIVE
  * @returns the normalized property value
  */
-export function normalizeObject(key: string[], value: any, cfnType?: string, pulumiProvider?: PulumiProvider): any {
+export function normalizePromptObject(key: string[], value: any, cfnType?: string, pulumiProvider?: PulumiProvider): any {
     if (!value) return value;
     if (Array.isArray(value)) {
         const result: any[] = [];
         for (let i = 0; i < value.length; i++) {
-            result[i] = normalizeObject(key, value[i], cfnType);
+            result[i] = normalizePromptObject(key, value[i], cfnType);
         }
         return result;
     }
 
-    if (typeof value !== 'object' || pulumi.Output.isInstance(value) || value instanceof Promise) {
```

Comment:
Could assert and fail if we see any of these:

 pulumi.Output.isInstance(value) || value instanceof Promise